### PR TITLE
Add `skip` support to `trigger` step

### PR DESCRIPTION
**Pull request by amartani**
This is supported by Buildkite's documentation: https://buildkite.com/docs/pipelines/trigger-step#trigger-step-attributes

### CODE REVIEW

**Comment by amartani** [comment]
@eleanorakh could you review this PR? (I'm not sure what's the process for sending PRs to the project; let me know if there is any other process that I should follow)

File: schema.json [hunk 1]
```diff
@@ -412,6 +412,18 @@
...
+        "anyOf": [
+          { "type": "boolean" },
+          { "type": "string" }
+        ],
+        "description": "Whether this step should be skipped. You can specify a reason for using a string.",
+        "examples": [
+          true,
+          false,
+          "My reason"
+        ]
```
Comment (pda) [inline schema.json:426]: This look legit 👍🏼 But I haven't tested it.

File: test/package-lock.json [hunk 2]
```diff
@@ -1,364 +1,6 @@
...
-      "dependencies": {
-        "punycode": "^2.1.0"
-      }
-    },
-    "node_modules/wrappy": {
-      "version": "1.0.2",
-      "resolved": "https://registry.npmjs.org/wrappy/-/wrappy-1.0.2.tgz",
-      "integrity": "sha1-tSQ9jz7BqjXxNkYFvA0QNuMKtp8="
-    }
-  },
```
Comment (pda) [inline test/package-lock.json:361]: I imagine this PR shouldn't change the `package-lock.json`?

File: test/valid-pipelines/trigger.yml [hunk 3]
```diff
@@ -76,3 +76,9 @@ steps:
 
   - trigger: "a-slug"
     allow_dependency_failure: true
+
+  - trigger: "a-slug"
+    skip: true
+
+  - trigger: "a-slug"
+    skip: "reason for skipping"
```
Comment (pda) [inline test/valid-pipelines/trigger.yml:84]: Probably unimportant, but there should be a newline at the end of the file.
I'm never sure if it's worth mentioning, but GitHub considers it important enough to visually flag it I guess.

**Review by pda — COMMENTED** [review]
Thanks for that @amartani 👍🏼
Looks good, a couple of questions / comments.

**Comment by eleanorakh** [comment]
Thanks so much for submitting @amartani, our apologies for the delay in reviewing it for you. There's a couple of tidy ups as pda suggested, then I'll merge this one in 😊

**Comment by pda** [comment]
Actually, looks like the tests are failing:

```
$ docker-compose run --rm tests
Creating test_tests_run ... done

> @ test /app/test
> mocha *.test.js



  schema.json
    ✓ should validate block steps (81ms)
    ✓ should validate input steps (42ms)
    ✓ should validate command steps
    ✓ should validate env blocks
    ✓ should validate blocks with extra properties
    1) should validate step groups
    ✓ should validate trigger steps
    ✓ should validate wait steps
    ✓ should validate notify


  8 passing (375ms)
  1 failing

  1) schema.json
       should validate step groups:
     AssertionError: expected 'data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should be string, data.steps[5] should be equal to one of the allowed values, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should be string, data.steps[5] should be equal to one of the allowed values, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should be string, data.steps[5] should be equal to one of the allowed values, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should match some schema in anyOf, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should be string, data.steps[6] should be equal to one of the allowed values, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should be string, data.steps[6] should be equal to one of the allowed values, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should be string, data.steps[6] should be equal to one of the allowed values, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should match some schema in anyOf, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should be string, data.steps[7] should be equal to one of the allowed values, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should be string, data.steps[7] should be equal to one of the allowed values, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should be string, data.steps[7] should be equal to one of the allowed values, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7].steps[0] should be equal to one of the allowed values, data.steps[7].steps[0] should be object, data.steps[7].steps[0] should be object, data.steps[7].steps[0] should be object, data.steps[7].steps[0] should be object, data.steps[7].steps[0] should match some schema in anyOf, data.steps[7].steps[1] should be string, data.steps[7].steps[1] should be equal to one of the allowed values, data.steps[7].steps[1].type should be equal to one of the allowed values, data.steps[7].steps[1] should NOT have additional properties, data.steps[7].steps[1] should NOT have additional properties, data.steps[7].steps[1].type should be equal to one of the allowed values, data.steps[7].steps[1] should NOT have additional properties, data.steps[7].steps[1] should NOT have additional properties, data.steps[7].steps[1] should match some schema in anyOf, data.steps[7].steps[2] should be string, data.steps[7].steps[2] should be equal to one of the allowed values, data.steps[7].steps[2] should NOT have additional properties, data.steps[7].steps[2] should NOT have additional properties, data.steps[7].steps[2] should NOT have additional properties, data.steps[7].steps[2] should NOT have additional properties, data.steps[7].steps[2] should match some schema in anyOf, data.steps[7] should match some schema in anyOf' to deeply equal undefined
      at validate (schema.test.js:17:48)
      at Context.<anonymous> (schema.test.js:38:5)



npm ERR! Test failed.  See above for more details.
ERROR: 1
```

Here's a word-wrapped truncated version of that very long line:

> AssertionError: expected 'data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should be string, data.steps[5] should be equal to one of the allowed values, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should be string, data.steps[5] should be equal to one of the allowed values, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should be string, data.steps[5] should be equal to one of the allowed values, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should match some schema in anyOf

I haven't looked into why it's failing yet.

File: test/valid-pipelines/trigger.yml (hunk 3 above)
Comment (amartani) [reply test/valid-pipelines/trigger.yml]: Fixed 👍

File: test/package-lock.json (hunk 2 above)
Comment (amartani) [reply test/package-lock.json]: Oops, not intended. I think this was changed when I ran tests with `npm install && npm test` as described in `README.md`. Anyway, I reverted the change to this file.

**Comment by amartani** [comment]
> Actually, looks like the tests are failing:
> 
> ```
> $ docker-compose run --rm tests
> Creating test_tests_run ... done
> 
> > @ test /app/test
> > mocha *.test.js
> 
> 
> 
>   schema.json
>     ✓ should validate block steps (81ms)
>     ✓ should validate input steps (42ms)
>     ✓ should validate command steps
>     ✓ should validate env blocks
>     ✓ should validate blocks with extra properties
>     1) should validate step groups
>     ✓ should validate trigger steps
>     ✓ should validate wait steps
>     ✓ should validate notify
> 
> 
>   8 passing (375ms)
>   1 failing
> 
>   1) schema.json
>        should validate step groups:
>      AssertionError: expected 'data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should be string, data.steps[5] should be equal to one of the allowed values, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should be string, data.steps[5] should be equal to one of the allowed values, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should be string, data.steps[5] should be equal to one of the allowed values, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should match some schema in anyOf, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should be string, data.steps[6] should be equal to one of the allowed values, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should be string, data.steps[6] should be equal to one of the allowed values, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should be string, data.steps[6] should be equal to one of the allowed values, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should NOT have additional properties, data.steps[6] should match some schema in anyOf, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should be string, data.steps[7] should be equal to one of the allowed values, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should be string, data.steps[7] should be equal to one of the allowed values, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should be string, data.steps[7] should be equal to one of the allowed values, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7] should NOT have additional properties, data.steps[7].steps[0] should be equal to one of the allowed values, data.steps[7].steps[0] should be object, data.steps[7].steps[0] should be object, data.steps[7].steps[0] should be object, data.steps[7].steps[0] should be object, data.steps[7].steps[0] should match some schema in anyOf, data.steps[7].steps[1] should be string, data.steps[7].steps[1] should be equal to one of the allowed values, data.steps[7].steps[1].type should be equal to one of the allowed values, data.steps[7].steps[1] should NOT have additional properties, data.steps[7].steps[1] should NOT have additional properties, data.steps[7].steps[1].type should be equal to one of the allowed values, data.steps[7].steps[1] should NOT have additional properties, data.steps[7].steps[1] should NOT have additional properties, data.steps[7].steps[1] should match some schema in anyOf, data.steps[7].steps[2] should be string, data.steps[7].steps[2] should be equal to one of the allowed values, data.steps[7].steps[2] should NOT have additional properties, data.steps[7].steps[2] should NOT have additional properties, data.steps[7].steps[2] should NOT have additional properties, data.steps[7].steps[2] should NOT have additional properties, data.steps[7].steps[2] should match some schema in anyOf, data.steps[7] should match some schema in anyOf' to deeply equal undefined
>       at validate (schema.test.js:17:48)
>       at Context.<anonymous> (schema.test.js:38:5)
> 
> 
> 
> npm ERR! Test failed.  See above for more details.
> ERROR: 1
> ```
> 
> Here's a word-wrapped truncated version of that very long line:
> 
> > AssertionError: expected 'data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should be string, data.steps[5] should be equal to one of the allowed values, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should be string, data.steps[5] should be equal to one of the allowed values, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should be string, data.steps[5] should be equal to one of the allowed values, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should NOT have additional properties, data.steps[5] should match some schema in anyOf
> 
> I haven't looked into why it's failing yet.

I couldn't reproduce it; the tests seem to pass on my machine, both locally with `npm install && npm test` and with docker-compose `docker-compose build && docker-compose run --rm tests`


```
> docker-compose build && docker-compose run --rm tests
Building tests
[+] Building 3.5s (10/10) FINISHED                                                     
 => [internal] load build definition from Dockerfile                              0.0s
 => => transferring dockerfile: 193B                                              0.0s
 => [internal] load .dockerignore                                                 0.0s
 => => transferring context: 2B                                                   0.0s
 => [internal] load metadata for docker.io/library/node:10                        0.0s
 => [1/5] FROM docker.io/library/node:10                                          0.1s
 => [internal] load build context                                                 1.4s
 => => transferring context: 5.20MB                                               1.3s
 => [2/5] WORKDIR /app/test                                                       0.0s
 => [3/5] COPY    package.json package-lock.json /app/test/                       0.0s
 => [4/5] RUN     npm install                                                     1.8s
 => [5/5] COPY    . /app/test                                                     0.1s
 => exporting to image                                                            0.1s 
 => => exporting layers                                                           0.1s 
 => => writing image sha256:26d2a266be61473081074fd0cd4d392f37fcb5d79da8dfc9ab76  0.0s 
 => => naming to docker.io/library/test_tests                                     0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
Creating test_tests_run ... done

> @ test /app/test
> mocha *.test.js



  schema.json
    ✓ should validate block steps (71ms)
    ✓ should validate input steps (42ms)
    ✓ should validate command steps
    ✓ should validate env blocks (41ms)
    ✓ should validate blocks with extra properties
    ✓ should validate step groups
    ✓ should validate trigger steps
    ✓ should validate wait steps
    ✓ should validate notify


  9 passing (339ms)
```

**Comment by amartani** [comment]
@pda could you take another look? Thank you!

**Comment by pda** [comment]
> I couldn't reproduce it; the tests seem to pass on my machine

Huh, me too!

Thanks for this.

**Pull request merged**